src/test/java/com/github/tomakehurst/wiremock/standalone/CommandLineOptionsTest.java

```diff
@@ -598,6 +601,42 @@ public void returnsTheSameInstanceOfTemplatingExtensionForEveryInterfaceImplemen
         assertSame(one, two);
     }
 
+    @Test
+    public void fileSourceDefaultsToSingleRootFileSource() {
+        CommandLineOptions options = new CommandLineOptions();
+
+        FileSource fileSource = options.filesRoot();
+
+        assertThat(fileSource, instanceOf(SingleRootFileSource.class));
+    }
+
+    @Test
+    public void mappingsSourceDefaultsToJsonFileMappingsSource() {
+        CommandLineOptions options = new CommandLineOptions();
+
+        MappingsSaver mappingsSaver = options.mappingsSaver();
+
+        assertThat(mappingsSaver, instanceOf(JsonFileMappingsSource.class));
+    }
+
+    @Test
+    public void rootDirUnderClasspathSetsFileSourceToUseClasspath() {
+        CommandLineOptions options = new CommandLineOptions("--root-dir-under-classpath");
+
+        FileSource fileSource = options.filesRoot();
+
+        assertThat(fileSource, instanceOf(ClasspathFileSource.class));
+    }
+
+    @Test
+    public void rootDirUnderClasspathSetsMappingsSourceToUseClasspath() {
+        CommandLineOptions options = new CommandLineOptions("--root-dir-under-classpath");
+
+        MappingsSaver mappingsSaver = options.mappingsSaver();
+
+        assertThat(mappingsSaver, instanceOf(JsonFileMappingsSource.class));
```

Comment:
:question: I'm not sure this is good enough as either way it's returned - thoughts?

Reply:
Is there anything stopping you calling `CommandLineOptions.filesRoot()` and checking the runtime class of the result?

Reply:
If you see above, we do have that tested - would you be happy that this is "good enough", as we're pretty confident that the mappings are sourced by the underlying `filesRoot`?